### ISSUE TITLE
Follow KDE Color Scheme

### ISSUE THREAD

**Issue by Zren** [PapirusDevelopmentTeam/papirus-icon-theme-kde#10]
Recently, the breeze icon theme now supports following the text color when drawing the icon. This is really cool when some windows are white on black, and others are black on white. I wrote a script to convert about 70% of the monochrome icons. It's not perfect and will need a human to fix the other 30%, but I'd figure I'd check how interested you'd be before touching most of the icons.

`convert_icon.py`
```python
#!/usr/bin/python3

from bs4 import BeautifulSoup
import sys

def insertStylesheet(soup):
	if not soup.svg.defs:
		defs = soup.new_tag('defs')
		soup.svg.insert(0, defs)
	
	if not soup.svg.style:
		style = soup.new_tag('style')
		style['id'] = 'current-color-scheme'
		style['type'] = 'text/css'
		style.string = '.ColorScheme-Text { color:#555555; }'
		soup.svg.defs.insert(0, style)

def convertIconSvgString(xml):
	soup = BeautifulSoup(xml, 'html.parser')
	#print(soup.svg.defs)

	modified = False
	paths = soup.find_all('path')
	for path in paths:
		# print(path['style'])
		if 'style' not in path and len(paths) == 1:
			path['style'] = 'fill:currentColor' # ;fill-opacity:1;stroke:none
		elif 'style' in path and 'fill:#555555' in path['style']:
			path['style'] = path['style'].replace('fill:#555555', 'fill:currentColor')
		else:
			continue

		del path['opacity']
		path['class'] = 'ColorScheme-Text'
		modified = True
	
	if soup.svg.style and 'type' not in soup.svg.style:
		soup.svg.style['type'] = 'text/css'
		modified = True

	if modified:
		insertStylesheet(soup)
		return soup.prettify(formatter="xml")
	else:
		return xml


def convertIconSvg(filename):
	with open(filename) as f:
		xml = f.read()

	xml2 = convertIconSvgString(xml)

	if xml != xml2:
		with open(filename, 'w') as f:
			f.write(xml2)

if __name__ == '__main__':
	print(sys.argv)
	convertIconSvg(sys.argv[1])
```

`convert_icontheme.py`
```python
#!/usr/bin/python3

from convert_icon import convertIconSvg
import glob
import os

for filename in glob.iglob('**/*.svg', recursive=True):
	print(filename)
	if os.path.islink(filename):
		continue
	convertIconSvg(filename)
```

Basically the script works by looking for a `#555` colour fill on a path, or if there's only 1 path, and converts that path to the color scheme text color.

It's not perfect, but it catches a large amount of the status icons.
![](https://i.imgur.com/BfKsJ7G.png)

**Comment by withthelemons** [comment]
This is great! Do you have repo so I can try it?

**Comment by Zren** [comment]
https://github.com/Zren/papirus-icon-theme-kde/archive/colorscheme.zip
https://github.com/Zren/papirus-icon-theme-kde/tree/colorscheme

I converted the `papirus` folder.

**Comment by FadeMind** [comment]
@Zren check mailbox. Please Create new branch for this changes in repo. Thanks.
